src/EnergyPlus/WeatherManager.cc

```diff
@@ -9476,20 +9475,31 @@ namespace WeatherManager {
                         }
                         lineIn = statFile.readLine();
                         lineAvg = lineIn.data;
+                        break;
                     }
                 }
-                int AnnualNumberOfDays = 0;
-                for (int i = 1; i <= 12; ++i) {
-                    MonthlyAverageDryBulbTemp(i) = OutputReportTabular::StrToReal(OutputReportTabular::GetColumnUsingTabs(lineAvg, i + 2));
-                    AnnualDailyAverageDryBulbTempSum += MonthlyAverageDryBulbTemp(i) * state.dataWeatherManager->EndDayOfMonth(i);
-                    MonthlyDailyDryBulbMin = min(MonthlyDailyDryBulbMin, MonthlyAverageDryBulbTemp(i));
-                    MonthlyDailyDryBulbMax = max(MonthlyDailyDryBulbMax, MonthlyAverageDryBulbTemp(i));
-                    AnnualNumberOfDays += state.dataWeatherManager->EndDayOfMonth(i);
+                // TODO: should we check if it has "Daily Avg" in it?
+                // if (lineAvg.find("Daily Avg") == std::string::npos)
+                if (lineAvg.empty()) {
+                    ShowSevereError(state,
+                                    "CalcAnnualAndMonthlyDryBulbTemp: Stat file '" + statFile.filePath.string() +
+                                        "' does not have Monthly Statistics for Dry Bulb temperatures.");
+                    ShowContinueError(state, "Water Mains Temperature will be set to a fixed default value of 10.0 C.");
+                    return;
```

Comment:
Some error checking, right now just checking that the line isn't empty (which it would be if "Monthly Statistics for dry bulb" wasn't found). Left a TODO. Thoughts @Myoldmopar 

TODO:
* [ ] address TODO

Reply:
The risk being that the search above to populate the `lineAvg` variable will somehow yield an invalid result?  I would say that if we need to verify the match on lineAvg, we should do it when we assign it.  If that's not the concern then please clarify.

Reply:
I think it should check if it has "Daily Avg" in the string. Just in case the stat file is malformed. Now whether it is done when you assign `lineAvg = lineIn.data;` above or where you have it now, I do not have strong opinions on that.

Reply:
My concern was whether checking for "Daily Avg" was correct or not... especially with respect to casing, whitespacing, etc.
I know we expect a line with 12 tab separated values, but do we know for sure all .stat files do have "Daily Avg" in front of it, spelled exactly like this?

Here is an example, the USA_AZ_Phoenix_TM2.stat
```
 - Monthly Statistics for Dry Bulb temperatures °C
 	           	Jan	Feb	Mar	Apr	May	Jun	Jul	Aug	Sep	Oct	Nov	Dec	
 	Maximum 	 26.1	 27.8	 32.8	 34.4	 41.1	 46.1	 45.0	 45.6	 40.6	 33.9	 32.8	 25.6	
 	 Day:Hour	28:15	28:16	17:15	 1:17	28:17	28:16	19:17	 1:17	 9:15	 2:17	12:14	 9:15	

 	Minimum 	  1.1	  0.0	  3.3	  6.1	 11.7	 16.1	 19.4	 22.8	 16.7	 10.6	  5.0	 -2.8	
 	 Day:Hour	20:07	23:07	 5:06	22:05	 7:05	16:05	 4:06	12:23	22:06	18:05	30:05	23:05	

 	Daily Avg	 12.1	 12.4	 18.5	 21.8	 25.9	 32.3	 33.3	 32.4	 29.6	 22.7	 17.3	 11.4	
```

Reply:
Given that the stat files are all generated by the EnergyPlus weather processor, it's safe to assume they will all start with "Daily Avg" on this line. And if the line doesn't start with that, then don't use it. If the stat file is present but you don't find this statistic, will it go ahead and calculate from the epw file?

Reply:
With the current proposed fix if not found, it will issue a (new) severe and return, not fallback to reading the EPW. (I think we could make that a fatal actually... If you have a broken .stat file, don't you want to be made aware?)